packages/client-readmodel-writer/src/clientConsumerServiceV1.ts

```diff
@@ -1,25 +1,150 @@
+              },
+            },
+          },
+          { upsert: true }
+        );
+      }
     )
+    .with({ type: "ClientPurposeAdded" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $set: {
+            "metadata.version": message.version,
+          },
+          $push: {
+            "data.purposes": message.data.statesChain?.purpose?.purposeId,
+          },
+        }
+      );
+    })
+    .with({ type: "ClientPurposeRemoved" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $pull: {
+            "data.purposes": message.data.purposeId,
+          },
+          $set: {
+            "metadata.version": message.version,
+          },
+        }
+      );
+    })
+    .with({ type: "RelationshipRemoved" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $pull: {
+            "data.relationships": message.data.relationshipId,
+          },
+          $set: {
+            "metadata.version": message.version,
+          },
+        }
+      );
+    })
+    .with({ type: "KeysAdded" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $push: {
+            "data.keys": {
+              $each: message.data.keys
+                .map((v) =>
+                  v.value ? toReadModelKey(fromKeyV1(v.value)) : undefined
+                )
+                .filter((k) => k !== undefined),
+            },
+          },
+          $set: {
+            "metadata.version": message.version,
+          },
+        }
+      );
+    })
+    .with({ type: "KeyDeleted" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $pull: {
+            "data.keys": { kid: message.data.keyId },
+          },
+          $set: {
+            "metadata.version": message.version,
+          },
+        }
+      );
+    })
+    .with({ type: "ClientDeleted" }, async (message) => {
+      await clients.deleteOne({
+        "data.id": message.stream_id,
+        "metadata.version": { $lt: message.version },
+      });
+    })
+    .with({ type: "KeyRelationshipToUserMigrated" }, async (message) => {
+      await clients.updateOne(
+        {
+          "data.id": message.stream_id,
+          "metadata.version": { $lt: message.version },
+        },
+        {
+          $push: {
+            "data.users": message.data.userId,
+          },
```

Comment:
`KeyRelationshipToUserMigrated` acts on `keys`, not `users`, see [here](https://github.com/pagopa/interop-be-authorization-management/blob/1.0.x/src/main/scala/it/pagopa/interop/authorizationmanagement/model/persistence/projection/ClientCqrsProjection.scala#L38)